src/state/slices/opportunitiesSlice/resolvers/uniV2/utils.test.ts

```diff
@@ -7,76 +7,55 @@ import BigNumber from 'bignumber.js'
 import type { IUniswapV2Pair } from 'contracts/abis/IUniswapV2Pair'
 import type { Address, GetContractReturnType, PublicClient, WalletClient } from 'viem'
 import { getContract } from 'viem'
+import { bn } from 'lib/bignumber/bignumber'
 import { TRADING_FEE_RATE } from 'state/slices/opportunitiesSlice/resolvers/uniV2/constants'
 import {
   calculateAPRFromToken0,
   getToken0Volume24Hr,
 } from 'state/slices/opportunitiesSlice/resolvers/uniV2/utils'
 
-jest.mock('contracts/contractManager', () => ({
-  getOrCreateContractByAddress: () => ({
-    filters: {
-      Swap: jest.fn(() => {}),
-    },
-    queryFilter: mockGetPastEvents,
-  }),
-  getOrCreateContractByType: () => ({
-    filters: {
-      Swap: jest.fn(() => {}),
-    },
-    queryFilter: mockGetPastEvents,
-  }),
-}))
-
-jest.mock('@ethersproject/contracts', () => ({
-  Contract: jest.fn().mockImplementation(() => ({
-    filters: {
-      Swap: jest.fn(() => {}),
-    },
-    queryFilter: mockGetPastEvents,
-  })),
-}))
-jest.mock('@uniswap/sdk', () => ({
-  Token: jest.fn().mockImplementation(),
-  TokenAmount: jest.fn().mockImplementation(() => ({
-    toFixed: () => tokenAmount,
-  })),
-}))
-
 const tokenAmount = '1000000'
-const amount0Out = '97000000000000000000000'
-const amount0In = '23000000000000000000000'
+const mockAmount0Out = '97000000000000000000000'
+const mockAmount0In = '23000000000000000000000'
 const blockNumber = 5000000
 
-const mockGetPastEvents = () =>
-  new Promise(resolve => {
-    resolve([
-      {
-        args: {
-          amount0Out,
-          amount1In: '100000000000000000000',
-        },
-      },
-      {
-        args: {
-          amount0In,
-          amount1Out: '223100000000000000000',
-        },
-      },
-    ])
-  })
+jest.mock('lib/viem-client', () => ({
+  viemEthMainnetClient: {
+    createEventFilter: jest.fn(() => ({})),
+    getFilterLogs: () =>
+      new Promise(resolve => {
+        resolve([
+          {
+            args: {
+              amount0Out: mockAmount0Out,
+              amount1In: '100000000000000000000',
+            },
+          },
+          {
+            args: {
+              amount0In: mockAmount0In,
+              amount1Out: '223100000000000000000',
+            },
+          },
+        ])
+      }),
+  },
+}))
 
 const mockContract = getContract({
   abi: [],
   address: '' as Address,
 }) as unknown as GetContractReturnType<typeof IUniswapV2Pair, PublicClient, WalletClient>
 const token0Decimals = 18
-const mockToken0Reserves = new TokenAmount(new Token(1, '', token0Decimals), tokenAmount)
+const mockToken0Reserves = new TokenAmount(
+  new Token(1, '0x0000000000000000000000000000000000000000', token0Decimals),
```

Comment:
Looks like this was never actually ran at runtime - this throws and expects a valid address